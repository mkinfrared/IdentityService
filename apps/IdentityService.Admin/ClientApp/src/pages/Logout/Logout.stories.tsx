import { Meta, StoryObj } from "@storybook/react";

import Logout from "./Logout";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/Logout",
  component: Logout,
  tags: ["autodocs"],
} satisfies Meta<typeof Logout>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <Logout {...args} />
    </div>
  ),
};

export { Default };

export default meta;
