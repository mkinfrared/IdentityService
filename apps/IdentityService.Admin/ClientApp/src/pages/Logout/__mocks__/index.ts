export { default } from "./Logout.mock";
