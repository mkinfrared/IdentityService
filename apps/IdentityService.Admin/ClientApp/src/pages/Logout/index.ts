export { default } from "./Logout";

export * from "./Logout.type";
