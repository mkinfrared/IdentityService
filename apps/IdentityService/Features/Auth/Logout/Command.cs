using MediatR;

namespace IdentityService.Features.Auth.Logout;

public partial class Logout
{
  public class Command : IRequest<string>
  {
    public Command(string logoutId)
    {
      LogoutId = logoutId;
    }

    public string LogoutId { get; }
  }
}
