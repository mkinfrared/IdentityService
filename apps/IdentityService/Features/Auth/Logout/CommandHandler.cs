using System.Threading;
using System.Threading.Tasks;

using IdentityServer4.Services;

using IdentityService.Entities;

using MediatR;

using Microsoft.AspNetCore.Identity;

namespace IdentityService.Features.Auth.Logout;

public partial class Logout
{
  public class CommandHandler : IRequestHandler<Command, string>
  {
    private readonly SignInManager<User> _signInManager;
    private readonly IIdentityServerInteractionService _interactionService;

    public CommandHandler(
      SignInManager<User> signInManager,
      IIdentityServerInteractionService interactionService
    )
    {
      _signInManager = signInManager;
      _interactionService = interactionService;
    }

    public async Task<string> Handle(Command request, CancellationToken cancellationToken)
    {
      await _signInManager.SignOutAsync();

      var logoutRequest = await _interactionService.GetLogoutContextAsync(request.LogoutId);

      return logoutRequest.PostLogoutRedirectUri;
    }
  }
}
