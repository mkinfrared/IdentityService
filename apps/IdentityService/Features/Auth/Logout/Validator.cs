using FluentValidation;

namespace IdentityService.Features.Auth.Logout;

public partial class Logout
{
    public class Validator : AbstractValidator<Command>
    {
        public Validator()
        {
            RuleFor(command => command.LogoutId).NotEmpty();
        }
    }
}
