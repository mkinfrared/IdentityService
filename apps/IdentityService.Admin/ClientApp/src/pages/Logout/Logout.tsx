import css from "./Logout.module.scss";
import { LogoutProps } from "./Logout.type";

const Logout = ({ className }: LogoutProps) => {
  const classNames = [css.Logout, className].filter(Boolean).join(" ");

  return (
    <div className={classNames} data-testid="Logout">
      Logout works!!
    </div>
  );
};

export default Logout;
