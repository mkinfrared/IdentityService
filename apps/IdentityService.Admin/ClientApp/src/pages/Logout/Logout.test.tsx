import { render } from "@testing-library/react";

import Logout from "./Logout";

describe("<Logout />", () => {
  const Component = <Logout />;

  it("should be defined", () => {
    expect(Logout).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Logout");

    expect(element).toBeDefined();
  });
});
