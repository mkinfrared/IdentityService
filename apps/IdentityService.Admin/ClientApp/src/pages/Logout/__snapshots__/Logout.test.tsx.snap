// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> should match the snapshot 1`] = `
<div>
  <div
    class="Logout"
    data-testid="Logout"
  >
    Logout works!!
  </div>
</div>
`;
